src/storage/store/column_chunk_data.cpp

```diff
@@ -113,14 +113,18 @@ void ColumnChunkData::initializeBuffer(common::PhysicalTypeID physicalType, Memo
 }
 
 void ColumnChunkData::initializeFunction(bool enableCompression) {
+    const auto compression = getCompression(dataType, enableCompression);
+    getMetadataFunction = GetCompressionMetadata(compression, dataType);
+    initializeFlushBufferFunction(compression);
```

Comment:
I suggest updating `initializeFlushBufferFunction` to return `flush_buffer_func_t` and change this to
```suggestion
    flushBufferFunction = initializeFlushBufferFunction(compression);
```
just for a bit more consistency and so we can remove all the `break` statements in `initializeFlushBufferFunction`

Reply:
Sounds good. Done now.